src/Relation/Nullary/Reflects.agda

```diff
@@ -0,0 +1,28 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of the `Reflects` construct
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Nullary.Reflects where
+
+open import Data.Bool.Base
+open import Level
+open import Relation.Nullary
+
+variable
+  p : Level
+  P : Set p
+
+------------------------------------------------------------------------
+-- `Reflects P b` is equivalent to `if b then P else ¬ P`.
+
+of : ∀ {b} → if b then P else ¬ P → Reflects P b
```

Comment:
I would probably call this `if-reflects` and the proof below `reflects-if`

Reply:
I think maybe the comment is what is wrong here, rather than the names. I only see these two functions being used when `b` is a value, so the user never actually sees an `if` in their goal. I have some examples of `invert` [here](https://github.com/agda/agda-stdlib/pull/929/commits/99b8729b7c830bb424cfeead3b341c2a7fc724e4#diff-3d30f647d768c855434815d99283082eR505-R519), where it is used to get the proof out of a `Reflects` value while being as unstrict as possible. We know, based on which branch we're in, whether `r` has type `Reflects (Q zero) false` or `Reflects (Q zero) true`, and we use `invert` to get out either a proof of `¬ Q zero` or of `Q zero`. I've also got an example [here](https://github.com/agda/agda-stdlib/pull/929/commits/99b8729b7c830bb424cfeead3b341c2a7fc724e4#diff-f720b1ed09bcf5d567bb95b6f63dafa1R49-R55) where I try it out mostly as a stylistic thing.

As for `of`, the potential use is to replace any instance of `ofʸ` or `ofⁿ` in expressions. Again, whether `does` is `true` or `false` will be known. A way to see it is that the `if` and the `b` control a form of overloading.

Reply:
Ah okay, that makes sense then. Thanks for the explanation!